src/OrchardCore.Modules/OrchardCore.Contents/Drivers/ContentsDriver.cs

```diff
@@ -44,34 +54,59 @@ public override IDisplayResult Display(ContentItem model, IUpdateModel updater)
                         ctx.Shape.Metadata.Alternates.Add($"{stereotype}__ContentsMetadata");
                     }
                 });
+
+                results.Add(contentsMetadataShape);
+
+                var hasViewPermission = await _authorizationService.AuthorizeAsync(context.User, CommonPermissions.ViewContent, contentItem);
+                var hasEditPermission = await _authorizationService.AuthorizeAsync(context.User, CommonPermissions.EditContent, contentItem);
+                var hasPublishPermission = await _authorizationService.AuthorizeAsync(context.User, CommonPermissions.PublishContent, contentItem);
+                var hasDeletePermission = await _authorizationService.AuthorizeAsync(context.User, CommonPermissions.DeleteContent, contentItem);
+                var hasPreviewPermission = await _authorizationService.AuthorizeAsync(context.User, CommonPermissions.PreviewContent, contentItem);
+                var hasClonePermission = await _authorizationService.AuthorizeAsync(context.User, CommonPermissions.CloneContent, contentItem);
```

Comment:
I wish we weren't having to do this in a `Display` method, but I don't see a better way :(

Reply:
I'll think about this one. Maybe there is a better way.